core/src/main/scala/algebra/ring/Ring.scala

```diff
@@ -74,4 +74,39 @@ trait RingFunctions[R[T] <: Ring[T]] extends AdditiveGroupFunctions[R] with Mult
 
 object Ring extends RingFunctions[Ring] {
   @inline final def apply[A](implicit ev: Ring[A]): Ring[A] = ev
+
+  /** Returns the given Double, understood as a rational number, in the provided
+    * (division) ring.
+    * 
+    * This is implemented in terms of basic ops. However, this is
+    * probably significantly less efficient than can be done with a specific
+    * type. So, it is recommended to specialize this general method.
+    */
+  def fromDouble[A](a: Double)(implicit ringA: Ring[A], mgA: MultiplicativeGroup[A]): A =
```

Comment:
Not sure I like this in `Ring`, since it can't actually be used with only a `Ring`. I think we should rename this to `defaultFromDouble` too, since we want to encourage folks to use the `fromDouble` on, eg, `Field`, which may be overridden to be faster.